packages/material-ui/src/Collapse/Collapse.js

```diff
@@ -41,153 +40,146 @@ export const styles = theme => ({
  * [Vertical Stepper](/demos/steppers/#vertical-stepper) StepContent component.
  * It uses [react-transition-group](https://github.com/reactjs/react-transition-group) internally.
  */
-class Collapse extends React.Component {
-  componentWillUnmount() {
-    clearTimeout(this.timer);
-  }
+const Collapse = React.forwardRef(function Collapse(props, ref) {
+  const {
+    children,
+    classes,
+    className,
+    collapsedHeight,
+    component: Component,
+    in: inProp,
+    onEnter,
+    onEntered,
+    onEntering,
+    onExit,
+    onExiting,
+    style,
+    theme,
+    timeout,
+    ...other
+  } = props;
+  const timer = React.useRef();
+  const wrapperRef = React.useRef();
+  const autoTransitionDuration = React.useRef();
 
-  handleEnter = node => {
-    node.style.height = this.props.collapsedHeight;
+  React.useEffect(() => {
+    return () => clearTimeout(timer.current);
```

Comment:
To make it clear we don't expect a returned value?
```suggestion
    return () => { clearTimeout(timer.current) };
```